app/javascript/packages/document-capture/components/address-search.tsx

```diff
@@ -12,37 +13,56 @@ interface Location {
 
 interface AddressSearchProps {
   onAddressFound?: (location: Location) => void;
+  registerField: (field: string) => {};
 }
 
 export const ADDRESS_SEARCH_URL = '/api/addresses';
 
-function AddressSearch({ onAddressFound = () => {} }: AddressSearchProps) {
+function AddressSearch({ onAddressFound = () => {}, registerField }: AddressSearchProps) {
+  const validatedFieldRef = useRef();
   const [unvalidatedAddressInput, setUnvalidatedAddressInput] = useState('');
   const [addressQuery, setAddressQuery] = useState({} as Location);
-  const handleAddressSearch = useCallback(async () => {
-    const addressCandidates = await request(ADDRESS_SEARCH_URL, {
-      method: 'POST',
-      headers: { 'Content-Type': 'application/json' },
-      json: { address: unvalidatedAddressInput },
-    });
-
-    const bestMatchedAddress = addressCandidates[0];
-    setAddressQuery(bestMatchedAddress);
-    onAddressFound(bestMatchedAddress);
-  }, [unvalidatedAddressInput]);
+
+  const handleAddressSearch = useCallback(
+    async (event) => {
+      event.preventDefault();
+      validatedFieldRef.current.reportValidity();
+      if (unvalidatedAddressInput === '') {
+        return null;
+      }
+      const addressCandidates = await request(ADDRESS_SEARCH_URL, {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        json: { address: unvalidatedAddressInput },
+      });
+
+      const bestMatchedAddress = addressCandidates[0];
+      setAddressQuery(bestMatchedAddress);
+      onAddressFound(bestMatchedAddress);
+    },
+    [unvalidatedAddressInput],
+  );
 
   return (
     <>
-      <TextInput
-        value={unvalidatedAddressInput}
-        onChange={(event: ChangeEvent) => {
-          const target = event.target as HTMLInputElement;
-
-          setUnvalidatedAddressInput(target.value);
-        }}
-        label="Search for an address"
-      />
-      <Button onClick={handleAddressSearch}>Search</Button>
+      <ValidatedField
+        ref={validatedFieldRef}
+        messages={{ valueMissing: 'Include a city, state, and ZIP code' }}
```

Comment:
Can we translate this text?

Reply:
Thanks, @allthesignals and @aduth, I made the suggested changes to `address-search.tsx` and `config/locales/in_person_proofing/en.yml `.